include/rticonnextdds-connector.h

```diff
@@ -8,39 +8,194 @@
-void RTIDDSConnector_setJSONInstance(void *connector, char *name, char *json_str);
-void* RTIDDSConnector_getWriter(void *connector, char *name);
-void* RTIDDSConnector_getReader(void *connector, char *name);
-void RTIDDSConnector_read(void *connector, char *name);
-void RTIDDSConnector_take(void *connector, char *name);
-double RTIDDSConnector_getSamplesLength(void *connector, char *name);
-double RTIDDSConnector_getInfosLength(void *connector, char *name);
-int RTIDDSConnector_getBooleanFromInfos(void *connector, char *name, int index, char *field_name);
+int RTI_Connector_get_sample_count(
+	void *self,
+	const char *entity_name,
+	double *value);
+
+int RTI_Connector_get_boolean_from_infos(
+	void *self,
+	int *return_value,
+	const char *entity_name,
+	int index,
+	const char *name);
+
+int RTI_Connector_set_json_instance(
+	void *self,
+	const char *entity_name,
+	const char *json);
+
+int RTI_Connector_set_boolean_into_samples(
+	void *self,
+	const char *entity_name,
+	const char *name,
+	int value);
+
+int RTI_Connector_set_number_into_samples(
+	void *self,
+	const char *entity_name,
+	const char *name,
+	double value);
+
+int RTI_Connector_set_string_into_samples(
+	void *self,
+	const char *entity_name,
+	const char *name,
+	const char *value);
 
 int RTI_Connector_get_json_from_infos(
-        void *self,
-        const char *entity_name,
-        int index,
-        const char *name,
-        char **return_value);
+	void *self,
+	const char *entity_name,
+	int index,
+	const char *name,
+	char **value);
 
-int RTI_Connector_get_boolean_from_infos(
-        void *self,
-        int *return_value,
-        const char *entity_name,
-        int index,
-        const char *name);
-
-void* RTIDDSConnector_getJSONSample(void *connector, char *name, int index);
-double RTIDDSConnector_getNumberFromSamples(void *connector, char *name, int index, char *field_name);
-int RTIDDSConnector_getBooleanFromSamples(void *connector, char *name, int index, char *field_name);
-void* RTIDDSConnector_getStringFromSamples(void *connector, char *name, int index, char *field_name);
-char* RTIDDSConnector_freeString(char *json);
-int RTIDDSConnector_wait(void *connector, int timeout);
+int RTI_Connector_get_json_sample(
+	void *self,
+	const char *entity_name,
+	int index,
+	char **json_str);
+
+int RTI_Connector_get_json_member(
+	void *self,
+	const char *entity_name,
+	int index,
+	const char *member_name,
+	char **json_str);
+
+int RTI_Connector_clear(
+	void *self,
+	const char *entity_name);
+
+int RTI_Connector_read(
+	void *self,
+	const char *entity_name);
+
+int RTI_Connector_take(
+	void *self,
+	const char *entity_name);
+
+int RTI_Connector_write(
+	void *self,
+	const char *entity_name,
+	const char *params_json);
+
+struct RTI_Connector_Options {
+        /* boolean */ int enable_on_data_event;
+	/* boolean */ int one_based_sequence_indexing;
```

Comment:
What is this comment?

Reply:
This comment means the attributes are boolean.